app/controllers/users_controller.rb

```diff
@@ -46,6 +46,35 @@ def check_account_creation_status
     end
   end
 
+  # Run a set of corrections on a user
+  def check
+    unless logged_in_as_admin?
```

Comment:
Hm, I would say that all these methods should be separate even if invoked with a single controller button, and I am inclined to say they should actually only be in the admin_users controller precisely because we _don't_ want them exposed except to admins going through our main interface -- james, because that controller is for admin actions on users it is more appropriate IMO than including them directly in the users controller and I think it's important to keep a strong separation for the admin-only actions. The button itself is only appearing under admin_users view anyway so it also makes more sense for it to be an admin_users action than a users action I think.